pkg/inject/sidecar.go

```diff
@@ -49,6 +49,10 @@ func Sidecar(jaeger *v1.Jaeger, dep *appsv1.Deployment) *appsv1.Deployment {
 		decorate(dep)
 
 		logFields.Debug("injecting sidecar")
+		// This is an update
+		if HasJaegerAgent(dep) {
```

Comment:
Alternatively, you could iterate over the existing containers, replacing the `jaeger-agent` if you find one, or appending a new one if there's none. This way, you don't need to get a new slice without the container, only to append the container after that.